packages/compass-generative-ai/src/components/ai-optin-modal.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+  Banner,
+  Body,
+  Link,
+  ConfirmationModal,
+  SpinLoader,
+  css,
+  spacing,
+  H3,
+  palette,
+} from '@mongodb-js/compass-components';
+import { AiImageBanner } from './ai-image-banner';
+import { closeOptInModal, optIn } from '../store/atlas-optin-reducer';
+import type { RootState } from '../store/atlas-ai-store';
+import { usePreference } from 'compass-preferences-model/provider';
+
+const GEN_AI_FAQ_LINK = 'https://www.mongodb.com/docs/generative-ai-faq/';
+
+type OptInModalProps = {
+  isOptInModalVisible: boolean;
+  isOptInInProgress: boolean;
+  onOptInModalClose: () => void;
+  onOptInClick: () => void;
+  projectId: string | undefined;
+};
+
+const titleStyles = css({
+  marginBottom: spacing[400],
+  marginTop: spacing[400],
+  marginLeft: spacing[500],
+  marginRight: spacing[500],
+  textAlign: 'center',
+});
+
+const bodyStyles = css({
+  marginBottom: spacing[400],
+  marginTop: spacing[400],
+  marginLeft: spacing[300],
+  marginRight: spacing[300],
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  textAlign: 'center',
+});
+
+const disclaimerStyles = css({
+  color: palette.gray.dark1,
+  marginTop: spacing[400],
+  marginLeft: spacing[800],
+  marginRight: spacing[800],
+});
+
+const bannerStyles = css({
+  padding: spacing[400],
+  marginTop: spacing[400],
+  textAlign: 'left',
+});
+const getButtonText = (isOptInInProgress: boolean) => {
+  return (
+    <>
+      &nbsp;Use Natural Language
+      {isOptInInProgress && (
+        <>
+          &nbsp;
+          <SpinLoader darkMode={true}></SpinLoader>
+        </>
+      )}
+    </>
+  );
+};
+
+export const AIOptInModal: React.FunctionComponent<OptInModalProps> = ({
+  isOptInModalVisible,
+  isOptInInProgress,
+  onOptInModalClose,
+  onOptInClick,
+  projectId,
+}) => {
+  const isProjectAIEnabled = usePreference('enableGenAIFeaturesAtlasProject');
+  const PROJECT_SETTINGS_LINK = projectId
```

Comment:
nit, fine as is too, we could also write this with the `URL()` format, like:
https://github.com/mongodb-js/compass/blob/b3eea8a6f5fdd323f8609eb1eaf75d4d38df30d8/packages/databases-collections/src/components/load-sample-data.tsx#L12

Reply:
would this be the same though since it converts it back to string form at the end of the function?

Reply:
Yeah I'm not sure if the URL constructor really does anything. I reckon no changes need 👍 
https://developer.mozilla.org/en-US/docs/Web/API/URL/URL